### ISSUE TITLE
This implementation does not has regularization?

### ISSUE THREAD

**Issue by zhujiem** [guoyang9/BPR-pytorch#3]

**Comment by guoyang9** [comment]
Check this line for weight decay or [l2 regularizaiton].(https://github.com/guoyang9/BPR-pytorch/blob/a21fbcde6ab85f5215fe0db9bdb9471116e1a667/main.py#L83)
